### PR TITLE
Cbs conflicts

### PR DESCRIPTION
**Pull request by BjoerklundJacob**

### CODE REVIEW

File: Simulation/Pathfinding/HighLevelCBS.cpp [hunk 1]
```diff
@@ -16,29 +21,37 @@ Solution HighLevelCBS::findSolution(std::shared_ptr<Graph> graph, std::vector<st
     /**
      * While OPEN not empty do
      */
+    int iterations = 0;
     while (open.size() > 0) {
```
Comment (Zzeeker) [inline Simulation/Pathfinding/HighLevelCBS.cpp:20]: ! open.empty() is probably faster

File: Simulation/Pathfinding/HighLevelCBS.cpp [hunk 2]
```diff
@@ -16,29 +21,37 @@ Solution HighLevelCBS::findSolution(std::shared_ptr<Graph> graph, std::vector<st
...
          */
+        // Error::log("finding Conflicts now:\n");
         std::vector<Conflict> conflicts = p->findConflicts();
         /**
          * If P has no conflicts then return P.solution
          */
         if (conflicts.size() == 0) {
+            Error::log("After ");
+            Error::log(std::to_string(iterations));
+            Error::log("iterations, we found a solution!\n");
```
Comment (Zzeeker) [inline Simulation/Pathfinding/HighLevelCBS.cpp]: Triple Error::log is slow and cursed

File: Simulation/Pathfinding/HighLevelCBS.cpp [hunk 3]
```diff
@@ -1,6 +1,11 @@
 #include "HighLevelCBS.hpp"
 
-Solution HighLevelCBS::findSolution(std::shared_ptr<Graph> graph, std::vector<std::shared_ptr<Agent>> agents, LowLevelCBS lowLevel){
+Solution HighLevelCBS::findSolution(std::shared_ptr<Graph> graph, std::vector<AgentInfo> agents, LowLevelCBS lowLevel){
+    for(auto a : agents){
+        Error::log("Goal: ");
+        Error::log(std::to_string(a.getGoal()->getId()));
+        Error::log("\n");
```
Comment (Zzeeker) [inline Simulation/Pathfinding/HighLevelCBS.cpp]: For loops with Error::log is cursed

**Review by Zzeeker — COMMENTED** [review]
Someone else take a quick look before merge

File: Simulation/Pathfinding/HighLevelCBS.cpp (hunk 1 above)
Comment (Zzeeker) [reply Simulation/Pathfinding/HighLevelCBS.cpp:20]: turns out empty is only faster if implementation is changed to lists, where its up to the implementation which is faster

**Review by Zzeeker — APPROVED** [review]

File: Simulation/Pathfinding/AgentInfo.cpp [hunk 4]
```diff
@@ -1,10 +1,36 @@
...
+AgentInfo::AgentInfo(int id, Action currentAction, std::shared_ptr<Vertex> destination){
+    this->id = id;
+    this->currentAction = currentAction;
+    this->goal = destination;
+}
+
+int AgentInfo::getId(){ return id; }
+Action AgentInfo::getCurrentAction(){ return currentAction; }
+std::shared_ptr<Vertex> AgentInfo::getGoal(){ return goal; }
+int AgentInfo::getTimeAtVertex(std::shared_ptr<Vertex> vertex){ return (vertex == goal) ? 300 : 20; }//TODO improve estimates
```
Comment (Simson475) [inline Simulation/Pathfinding/AgentInfo.cpp:30]: and this

File: Simulation/Pathfinding/Agent.cpp [hunk 5]
```diff
@@ -32,7 +32,7 @@ Action Agent::getCurrentAction(){
 }
 
 int Agent::getTimeAtVertex(std::shared_ptr<Vertex> vertex){
-    return (vertex->getId() == this->bot->getStationPlan().front()) ? 10 : 1;//TODO actual times!
+    return (vertex->getId() == this->bot->getStationPlan().front()) ? 100 : 20;//TODO actual times! Also is this dublicate from deltas in conflict?
```
Comment (Simson475) [inline Simulation/Pathfinding/Agent.cpp]: streamline this

File: Simulation/Pathfinding/AgentInfo.hpp [hunk 6]
```diff
@@ -2,20 +2,33 @@
 #define AGENT_INFO_HPP
 
 #include "Location.hpp"
+#include "Action.hpp"
+#include "Vertex.hpp"
 #include <memory>
-#include "HighLevelCBS.hpp"
+
+/**
```
Comment (Simson475) [inline Simulation/Pathfinding/AgentInfo.hpp]: fjern?

File: Simulation/Pathfinding/ConstraintTree/ConstraintTree.cpp [hunk 7]
```diff
@@ -24,13 +24,104 @@ Solution ConstraintTree::getSolution(){
...
+                                    cStart = a2Start - delta;
+                                    cEnd = a2Start;
+                                }
+                                else {
+                                    cStart = a1Start - delta;
+                                    cEnd = a1Start;
+                                }
+                                conflicts.push_back(Conflict(conflictAgents, cStart, cEnd,
+                                    Location(ELocationType::VERTEX_LOCATION, a1.startVertex)));
+                                continue;
```
Comment (Simson475) [inline Simulation/Pathfinding/ConstraintTree/ConstraintTree.cpp:85]: fjern alle continues

File: Simulation/Pathfinding/ExperimentData.cpp [hunk 8]
```diff
@@ -49,4 +49,14 @@ void ExperimentData::distributeSolution(Solution solution){
     for (auto agent : getAgents()){
         agent->getBot()->setPath(solution.paths[agent->getId()]);
     }
+}
+
+std::vector<AgentInfo> ExperimentData::getAgentsInfo(){
+    auto agents = getAgents();
+    int size = agents.size();
```
Comment (Simson475) [inline Simulation/Pathfinding/ExperimentData.cpp:56]: ryk den ind i for loopen

File: Simulation/Pathfinding/ConstraintTree/ConstraintTree.cpp (hunk 7 above)
Comment (BjoerklundJacob) [reply Simulation/Pathfinding/ConstraintTree/ConstraintTree.cpp:85]: Løses i chained MR

**Pull request merged**